libfwupdplugin/fu-security-attrs.c

```diff
@@ -204,7 +204,7 @@ fu_security_attrs_calculate_hsi(FuSecurityAttrs *self, FuSecurityAttrsFlags flag
 {
 	guint hsi_number = 0;
 	FwupdSecurityAttrFlags attr_flags = FWUPD_SECURITY_ATTR_FLAG_NONE;
-	GString *str = g_string_new("HSI:");
+	g_autoptr(GString) str = g_string_new("HSI:");
```

Comment:
is this right?  I thought that how `g_string_free` was used is proper.

Reply:
Ya, we need to use the slightly weird `g_string_free(g_steal_pointer(&str), FALSE)` to avoid a double-free. The PVS linter was getting upset that we could return_val_if_fail and leak the string -- although we probably have bigger problems at that point...

Reply:
I guess you could do g_string_free in `g_return_val_if_fail`, but yeah that seems weird too.